libraries/AP_NavEKF/AP_NavEKF_Source.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <AP_Param/AP_Param.h>
+
+#define AP_NAKEKF_SOURCE_COUNT 3    // three banks of sources
+
+class AP_NavEKF_Source
+{
+
+public:
+    // Constructor
+    AP_NavEKF_Source();
+
+    /* Do not allow copies */
+    AP_NavEKF_Source(const AP_NavEKF_Source &other) = delete;
+    AP_NavEKF_Source &operator=(const AP_NavEKF_Source&) = delete;
+
+    enum class SourceXY {
+        NONE = 0,
+        // BARO = 1 (not applicable)
+        // RANGEFINDER = 2 (not applicable)
+        GPS = 3,
+        BEACON = 4,
+        OPTFLOW = 5,
+        EXTNAV = 6,
+        WHEEL_ENCODER = 7
+    };
+
+    enum class SourceZ {
+        NONE = 0,
+        BARO = 1,
+        RANGEFINDER = 2,
+        GPS = 3,
+        BEACON = 4,
+        // OPTFLOW = 5 (not applicable)
+        EXTNAV = 6
+    };
+
+    enum class SourceYaw {
+        NONE = 0,
+        COMPASS = 1,
+        EXTERNAL = 2,
+        EXTERNAL_COMPASS_FALLBACK = 3
+    };
```

Comment:
I would also like to separate out the "EXTERNAL_" into the various options (ExternalNav, GPS-for-Yaw (i.e. two GPSs), Yaw-directly-from-GPS, GSF) but this will require discussion with Tridge, Paul, etc and in any case is too big a change to try and include in this already large PR.  I suppose if we know that we are going to do that for sure we could possibly make it easier for ourselves in the future by setting "EXTERNAL" = "9" and EXTNAL_COMPASS_FALLBACK = "10".  This would let us use "3" for "GPS" later but on the other hand we already have two different methods of getting yaw from GPS so a single enum may not be enough.

@tridge any thoughts on this?